File: app/src/main/java/com/verbatoria/di/schedule/ScheduleModule.java
package com.verbatoria.di.schedule;

import com.verbatoria.business.schedule.IScheduleInteractor;
import com.verbatoria.business.schedule.ScheduleInteractor;
import com.verbatoria.data.repositories.schedule.IScheduleRepository;
import com.verbatoria.data.repositories.schedule.ScheduleRepository;
import com.verbatoria.ui.schedule.presenter.ISchedulePresenter;
import com.verbatoria.ui.schedule.presenter.SchedulePresenter;

import dagger.Module;
import dagger.Provides;

/**
 * Модуль даггера для расписания
 *
 * @author nikitaremnev
 */
@Module
public class ScheduleModule {

    @Provides
    @ScheduleScope
    IScheduleRepository provideScheduleRepository() {
        return new ScheduleRepository();
    }

    @Provides
    @ScheduleScope
    IScheduleInteractor provideScheduleInteractor(IScheduleRepository scheduleRepository) {
        return new ScheduleInteractor(scheduleRepository);
    }

    @Provides
    @ScheduleScope
    ISchedulePresenter provideSchedulePresenter(IScheduleInteractor scheduleInteractor) {
        return new SchedulePresenter(scheduleInteractor);
    }

}

File: app/src/main/java/com/verbatoria/di/schedule/ScheduleModule.kt
package com.verbatoria.di.schedule

import com.verbatoria.business.schedule.ScheduleInteractorImpl
import com.verbatoria.domain.schedule.ScheduleManager
import com.verbatoria.infrastructure.rx.RxSchedulersFactory
import com.verbatoria.ui.schedule.SchedulePresenter
import dagger.Module
import dagger.Provides
import dagger.Reusable

/**
 * @author n.remnev
 */

@Module
class ScheduleModule {

    @Provides
    @Reusable
    fun provideSchedulePresenter(
        scheduleManager: ScheduleManager,
        schedulersFactory: RxSchedulersFactory
    ): SchedulePresenter =
        SchedulePresenter(
            ScheduleInteractorImpl(
                scheduleManager,
                schedulersFactory
            )
        )

}